### PR TITLE
feat(client): add tests/ gift card API/ multiple fixes

### LINKED ISSUE #1099: The AsyncClient.futures_place_batch_order provide an "1022 INVALID_SIGNATURE" error
**Describe the bug**
The AsyncClient.futures_place_batch_order provide an "1022 INVALID_SIGNATURE" error

**To Reproduce**
```

from binance.client import AsyncClient, Client
import asyncio

api_key = ''
api_secret = ''

async def main():
    client_1_a = AsyncClient(api_key, api_secret)
    client_1_s = Client(api_key, api_secret)

    params_in = {"batchOrders": []}
    params1 = {
            "symbol": "ADAUSDT",
            "side": "BUY",
            "positionSide": "LONG",
            "type": "LIMIT",
            "quantity": "20",
            "price": "1.40",
            "timeInForce": "GTC"
            }
    params2 = {
            "symbol": "CHRUSDT",
            "side": "BUY",
            "positionSide": "LONG",
            "type": "LIMIT",
            "quantity": "30",
            "price": "0.63",
            "timeInForce": "GTC"
            }
    params3 = {
            "symbol": "HOTUSDT",
            "side": "BUY",
            "positionSide": "LONG",
            "type": "LIMIT",
            "quantity": "2000",
            "price": "0.0095",
            "timeInForce": "GTC"
            }
    params_in["batchOrders"].append(params1)
    params_in["batchOrders"].append(params2)
    params_in["batchOrders"].append(params3)
    
    # Sequential future order and batch order working
    client_1_s.futures_create_order(**params1)
    client_1_s.futures_place_batch_order(**params_in)

    # Normal async future order working
    await client_1_a.futures_create_order(**params1)

    # async future batch order NOT working 
    await client_1_a.futures_place_batch_order(**params_in)
    # binance.exceptions.BinanceAPIException: APIError(code=-1022): Signature for this request is not valid.

    await client_1_a.session.close()

if __name__ == "__main__":
    asyncio.run(main())

```

**Expected behavior**
 A working order without signature error

**Environment (please complete the following information):**
 python3.7 / python3.9
 debian 10 / Windows 10
python-binance tested on all major version from 1.0.0
New API keys with with spot and future enable just to be sure

**Logs or Additional context**

### LINKED ISSUE #1472: Wish to add a gift card redemption interface
https://www.binance.com/zh-CN/gift-card

### PR DESCRIPTION
**Pull request by pcriadoperez**
Dependent on PR #1464
- Adds tests. By default does not run margin or portfolio or private options tests. (Can activate with flag --run-margin and --run-portfolio) (This endpoints as are live were not tested)
- Adds missing endpoint for futures_edit_order
- Fixes error for some get requests missing ContentType in the header for async requests
- Fix batch orders async call
- Requires to whitelist eapi.binance.com in the proxy
- Add pytest-xdist to run tests in parallel
- Adds gift card api
- activate ruff format
- add missing futures_modify_order from async client
fixes #1099 #1474 
fixes #1472

### CODE REVIEW

File: binance/base_client.py [hunk 1]
```diff
@@ -206,6 +206,7 @@ def __init__(
     def _get_headers(self) -> Dict:
         headers = {
             "Accept": "application/json",
+            "Content-Type": "application/json",
```
Comment (carlosmiei) [inline binance/base_client.py:209]: interesting at ccxt we're using `'Content-Type': 'application/x-www-form-urlencoded',`

**Comment by carlosmiei** [comment]
DEMO swap

```
python examples/create_order_async.py
{'orderId': 4067449380, 'symbol': 'BTCUSDT', 'status': 'NEW', 'clientOrderId': 'x-Cb7ytekJ37b8cfbafacdfa36cd379', 'price': '0.00', 'avgPrice': '0.00', 'origQty': '0.010', 'executedQty': '0.000', 'cumQty': '0.000', 'cumQuote': '0.00000', 'timeInForce': 'GTC', 'type': 'MARKET', 'reduceOnly': True, 'closePosition': False, 'side': 'SELL', 'positionSide': 'LONG', 'stopPrice': '0.00', 'workingType': 'CONTRACT_PRICE', 'priceProtect': False, 'origType': 'MARKET', 'priceMatch': 'NONE', 'selfTradePreventionMode': 'NONE', 'goodTillDate': 0, 'updateTime': 1732204122751}
{'orderId': 4067449380, 'symbol': 'BTCUSDT', 'status': 'FILLED', 'clientOrderId': 'x-Cb7ytekJ37b8cfbafacdfa36cd379', 'price': '0.00', 'avgPrice': '99971.20000', 'origQty': '0.010', 'executedQty': '0.010', 'cumQuote': '999.71200', 'timeInForce': 'GTC', 'type': 'MARKET', 'reduceOnly': True, 'closePosition': False, 'side': 'SELL', 'positionSide': 'LONG', 'stopPrice': '0.00', 'workingType': 'CONTRACT_PRICE', 'priceProtect': False, 'origType': 'MARKET', 'priceMatch': 'NONE', 'selfTradePreventionMode': 'NONE', 'goodTillDate': 0, 'time': 1732204122751, 'updateTime': 1732204122751}
[{'symbol': 'LTCUSDT', 'positionSide': 'LONG', 'positionAmt': '3.300', 'entryPrice': '78.63030303029', 'breakEvenPrice': '78.65361575758', 'markPrice': '88.28000000', 'unRealizedProfit': '31.84400000', 'liquidationPrice': '0', 'isolatedMargin': '0', 'notional': '291.32400000', 'marginAsset': 'USDT', 'isolatedWallet': '0', 'initialMargin': '29.13240000', 'maintMargin': '1.89360600', 'positionInitialMargin': '29.13240000', 'openOrderInitialMargin': '0', 'adl': 0, 'bidNotional': '0', 'askNotional': '0', 'updateTime': 1732119287099}, {'symbol': 'BTCUSDT', 'positionSide': 'LONG', 'positionAmt': '0.015', 'entryPrice': '79888.804', 'breakEvenPrice': '66580.45818932', 'markPrice': '96489.20000000', 'unRealizedProfit': '249.00594000', 'liquidationPrice': '0', 'isolatedMargin': '0', 'notional': '1447.33800000', 'marginAsset': 'USDT', 'isolatedWallet': '0', 'initialMargin': '289.46760000', 'maintMargin': '5.78935200', 'positionInitialMargin': '289.46760000', 'openOrderInitialMargin': '0', 'adl': 0, 'bidNotional': '0', 'askNotional': '0', 'updateTime': 1732204122751}]
[{'accountAlias': 'AuoCAuuXmYFz', 'asset': 'FDUSD', 'balance': '0.00000000', 'crossWalletBalance': '0.00000000', 'crossUnPnl': '0.00000000', 'availableBalance': '0.00000000', 'maxWithdrawAmount': '0.00000000', 'marginAvailable': True, 'updateTime': 0}, {'accountAlias': 'AuoCAuuXmYFz', 'asset': 'BTC', 'balance': '0.00000000', 'crossWalletBalance': '0.00000000', 'crossUnPnl': '0.00000000', 'availableBalance': '0.00000000', 'maxWithdrawAmount': '0.00000000', 'marginAvailable': True, 'updateTime': 0}, {'accountAlias': 'AuoCAuuXmYFz', 'asset': 'BNB', 'balance': '0.00000000', 'crossWalletBalance': '0.00000000', 'crossUnPnl': '0.00000000', 'availableBalance': '0.00000000', 'maxWithdrawAmount': '0.00000000', 'marginAvailable': True, 'updateTime': 0}, {'accountAlias': 'AuoCAuuXmYFz', 'asset': 'ETH', 'balance': '0.00000000', 'crossWalletBalance': '0.00000000', 'crossUnPnl': '0.00000000', 'availableBalance': '0.00000000', 'maxWithdrawAmount': '0.00000000', 'marginAvailable': True, 'updateTime': 0}, {'accountAlias': 'AuoCAuuXmYFz', 'asset': 'USDT', 'balance': '4517.42752243', 'crossWalletBalance': '4517.42752243', 'crossUnPnl': '280.84994000', 'availableBalance': '4475.45346243', 'maxWithdrawAmount': '4475.45346243', 'marginAvailable': True,
```

DEMO Spot

```
{'symbol': 'BTCUSDT', 'orderId': 6441767, 'orderListId': -1, 'clientOrderId': 'x-HNA2TXFJ969f76bc7d0b68c0a2076d', 'transactTime': 1732204206772, 'price': '0.00000000', 'origQty': '0.01000000', 'executedQty': '0.01000000', 'cummulativeQuoteQty': '961.93960000', 'status': 'FILLED', 'timeInForce': 'GTC', 'type': 'MARKET', 'side': 'SELL', 'workingTime': 1732204206772, 'fills': [{'price': '96193.96000000', 'qty': '0.01000000', 'commission': '0.00000000', 'commissionAsset': 'USDT', 'tradeId': 2320696}], 'selfTradePreventionMode': 'EXPIRE_MAKER'}
{'symbol': 'BTCUSDT', 'orderId': 6441767, 'orderListId': -1, 'clientOrderId': 'x-HNA2TXFJ969f76bc7d0b68c0a2076d', 'price': '0.00000000', 'origQty': '0.01000000', 'executedQty': '0.01000000', 'cummulativeQuoteQty': '961.93960000', 'status': 'FILLED', 'timeInForce': 'GTC', 'type': 'MARKET', 'side': 'SELL', 'stopPrice': '0.00000000', 'icebergQty': '0.00000000', 'time': 1732204206772, 'updateTime': 1732204206772, 'isWorking': True, 'workingTime': 1732204206772, 'origQuoteOrderQty': '0.00000000', 'selfTradePreventionMode': 'EXPIRE_MAKER'}
{'asset': 'USDT', 'free': '10854.94831000', 'locked': '0.00000000'}
(.binance) ➜  python-binance git:(all-tests) ✗ 
```

File: binance/base_client.py (hunk 1 above)
Comment (pcriadoperez) [reply binance/base_client.py:209]: Interesting, yes, I would get it in test_options_price and test_options_mark_price

Comment (pcriadoperez) [reply binance/base_client.py:209]: according to the docs: 
```
For POST, PUT, and DELETE endpoints, the parameters may be sent as a query string or in the request body with content type application/x-www-form-urlencoded
```

i am wondering if we should remove this.. without it doesn't pass the tests.. but not sure if maybe with it it's failing and we just aren't seeing it in one of the skipped tests

Comment (pcriadoperez) [reply binance/base_client.py:209]: Just in case I edited the headers to be more similar to ccxt's implementation and what the docs say

Comment (carlosmiei) [reply binance/base_client.py:209]: @pcriadoperez Should we also set it for get requests?

**Pull request merged**